js/plates-model/model.ts

```diff
@@ -77,10 +80,40 @@ export default class Model {
     model.stepIdx = props.stepIdx;
     model.lastPlateDivisionOrMerge = props.lastPlateDivisionOrMerge;
     model.plates = props.plates.map((serializedPlate: ISerializedPlate) => Plate.deserialize(serializedPlate));
+    model.nextPlateId = props.nextPlateId || Math.max(...model.plates.map(p => p.id)) + 1;
     model.calculateDynamicProperties(false);
     return model;
   }
 
+  getNextPlateId() {
+    return this.nextPlateId++;
+  }
+
+  getNextPlateHue(id: number, hueToAvoid?: number) {
+    const getRandomItem = (set: Set<number>) => {
+      const items = Array.from(set);
+      return items[Math.floor(Math.random() * items.length)];
+    };
+    const availableHues = new Set<number>(plateHues);
+
+    if (id < plateHues.length) {
+      // Use color that matches the plate id.
+      return plateHues[id];
+    }
+    if (this.plates.length < plateHues.length) {
+      // Return first color that is not used yet.
+      this.plates.forEach(p => {
+        availableHues.delete(p.hue);
+      });
+      return getRandomItem(availableHues);
+    }
+    // Very unlikely, but if all the available hues are used, return random color.
+    if (hueToAvoid !== undefined) {
+      availableHues.delete(hueToAvoid);
+    }
+    return getRandomItem(availableHues);
```

Comment:
Is random better than something like `availableHues[id % availableHues.length]`? Seems like there might be some value to a consistently repeating pattern.

Reply:
This is the most unlikely use case when we already have 10 plates, and 11th is added (I don't think that's possible in practice with current settings).